pkg/oci/repository.go

```diff
@@ -51,16 +56,105 @@ type ResourceOptions struct {
 	SnapshotName     string
 }
 
+// ClientOptsFunc options are used to leave the cache backwards compatible.
+// If the certificate isn't defined, we will use `WithInsecure`.
+type ClientOptsFunc func(opts *Client)
+
+// WithCertificateLocation sets up certificates for the client.
+func WithCertificateLocation(name string) ClientOptsFunc {
+	return func(opts *Client) {
+		opts.CertificateLocation = name
+	}
+}
+
+// WithNamespace sets up certificates for the client.
+func WithNamespace(namespace string) ClientOptsFunc {
+	return func(opts *Client) {
+		opts.Namespace = namespace
+	}
+}
+
+// WithInsecureSkipVerify sets up certificates for the client.
+func WithInsecureSkipVerify(value bool) ClientOptsFunc {
+	return func(opts *Client) {
+		opts.InsecureSkipVerify = value
+	}
+}
+
+// WithClient sets up certificates for the client.
+func WithClient(client client.Client) ClientOptsFunc {
+	return func(opts *Client) {
+		opts.Client = client
+	}
+}
+
 // Client implements the caching layer and the OCI layer.
 type Client struct {
-	OCIRepositoryAddr string
+	Client              client.Client
+	OCIRepositoryAddr   string
+	CertificateLocation string
+	InsecureSkipVerify  bool
+	Namespace           string
+
+	serverPem    []byte
+	serverKeyPem []byte
```

Comment:
```suggestion
	key []byte
```

Those are used in client side right?